src/client/renderingengine.cpp

```diff
@@ -186,23 +187,119 @@ bool RenderingEngine::print_video_modes()
 	return videomode_list != NULL;
 }
 
-void RenderingEngine::setXorgClassHint(
-		const video::SExposedVideoData &video_data, const std::string &name)
+bool RenderingEngine::setupTopLevelWindow(const std::string &name)
+{
+	// FIXME: It would make more sense for there to be a switch of some
+	// sort here that would call the correct toplevel setup methods for
+	// the environment Minetest is running in but for now not deviating
+	// from the original pattern.
+	
+	/* Setting Xorg properties for the top level window */
+	setupTopLevelXorgWindow(name);
+	/* Done with Xorg properties */
+
+	/* Setting general properties for the top level window */
+	verbosestream << "Client: Configuring general top level"
+		<< " window properties"
+		<< std::endl;
+
+	bool result = setWindowIcon();
+
+	verbosestream << "Client: Finished configuring general top level"
+		<< " window properties"
+		<< std::endl;
+	/* Done with general properties */
+
+	// FIXME: setWindowIcon returns a bool result but it is unused.
+	// For now continue to return this result. 
+	return result;
+}
+
+void RenderingEngine::setupTopLevelXorgWindow(const std::string &name)
 {
 #ifdef XORG_USED
-	if (video_data.OpenGLLinux.X11Display == NULL)
+	const video::SExposedVideoData exposedData = driver->getExposedVideoData();
+
+	Display *x11_dpl = (Display *)exposedData.OpenGLLinux.X11Display;
+	if (x11_dpl == NULL) {
+		warningstream << "Client: Could not find X11 Display in ExposedVideoData"
+			<< std::endl; 
 		return;
+	}
+
+	verbosestream << "Client: Configuring Xorg specific top level"
+		<< " window properties"
+		<< std::endl;
+
 
+	Window x11_win = (Window)exposedData.OpenGLLinux.X11Window;
```

Comment:
same

Reply:
This one may not be necessary. I will try to compile it without it.